sql/aiven_extras.sql

```diff
@@ -521,7 +523,22 @@ BEGIN
         FALSE
     ) THEN
         RAISE EXCEPTION 'Configuring superuser roles not allowed: %', arg_role;
-    ELSIF arg_parameter NOT IN (
+    END IF;
+    WITH RECURSIVE tree AS (
+    -- Start with the possibly admin_role
+    SELECT member AS base_role, roleid, admin_option, member FROM pg_auth_members
```

Comment:
Shouldn't the `WHERE base_role = session_user::regrole::oid` be applied in the initial set? At least if I read this right, you start with all rows in pg_auth_members and then add even more rows to it by adding the rows again which have members, only to drop all rows which are not having the above condition.